qemu/tests/blockdev_snapshot_merge.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from provider.blockdev_snapshot_base import BlockDevSnapshotTest
+
+
+class BlockdevSnapshotMergeTest(BlockDevSnapshotTest):
+    def pre_test(self):
+        if not self.main_vm.is_alive():
+            self.main_vm.create()
+        self.main_vm.verify_alive()
+        if self.base_tag != "image1":
+            self.configure_data_disk()
+
+    def prepare_snapshot_file(self):
+        self.params["image_size_%s" % self.snapshot_tag] = self.base_image.size
+        self.params["image_name_%s" % self.snapshot_tag] = 'images/' + self.snapshot_tag
+        snapshot_format = self.params.get("snapshot_format", "qcow2")
+        self.params["image_format_%s" % self.snapshot_tag] = snapshot_format
+        super(BlockdevSnapshotMergeTest, self).prepare_snapshot_file()
+
+    def verify_snapshot(self):
+        if self.main_vm.is_alive():
+            self.main_vm.destroy(free_mac_addresses=False)
+        self.snapshot_image_list = []
+        snapshot = self.snapshot_tags[-1]
+        for item in self.snapshot_tags[-2::-1]:
+            base_image = self.get_image_by_tag(item)
+            snapshot_image = self.get_image_by_tag(snapshot)
+            self.snapshot_image_list.append(snapshot_image)
```

Comment:
self.snapshot_image_list.append(x) should be called right after prepare_snapshot_file(), consider this situation, something wrong happened before verify_snapshot(), snapshot_image_list is empty, no snapshots will be removed.

Reply:
Have updated it, thanks zhenchao.